### PR TITLE
build: bump golangci-lint version in build image

### PR DESCRIPTION
**Pull request by hairyhum**
## Change Overview

Old version of golangci-lint has a memory leak causing it to crash with oom in recent go versions.
Updated golangci-lint version in build image.
Fixed lint errors.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [x] :building_construction: Build

## Issues 

- fixes #issue-number

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/function/delete_rds_snapshot.go [hunk 1]
```diff
@@ -55,22 +55,22 @@ func (*deleteRDSSnapshotFunc) Name() string {
 	return DeleteRDSSnapshotFuncName
 }
 
-func deleteRDSSnapshot(ctx context.Context, snapshotID string, profile *param.Profile, dbEngine RDSDBEngine) (map[string]interface{}, error) {
```
Comment (KastenMike) [inline pkg/function/delete_rds_snapshot.go:58]: Should this be part of this PR?

Comment (hairyhum) [reply pkg/function/delete_rds_snapshot.go:58]: This fixes an `unparam` linter warning.

Comment (julio-lopez) [reply pkg/function/delete_rds_snapshot.go:58]: @KastenMike I am assuming this was flagged by a linter as part of the upgrade. @hairyhum can confirm.

If that's the case, there are 2 options:
- Put all the changes in a single PR, like this one; or
- 2 PRs in this order: the first one addressing the linter warnings, and the second one upgrading the linter version.

File: pkg/function/create_volume_snapshot.go [hunk 2]
```diff
@@ -134,7 +134,7 @@ func createVolumeSnapshot(ctx context.Context, tp param.TemplateParams, cli kube
 	}
 	wg.Wait()
 
-	err := fmt.Errorf(strings.Join(errstrings, "\n"))
+	err := fmt.Errorf("%s", strings.Join(errstrings, "\n"))
```
Comment (julio-lopez) [inline pkg/function/create_volume_snapshot.go:137]: It seems that this should use `errors.New`

```suggestion
	err := errors.New(strings.Join(errstrings, "\n"))
```

Question unrelated to this PR why are newlines (`\n`) being used inside the error message?

**Review by julio-lopez — APPROVED** [review]
LG. See minor comment

File: pkg/function/create_volume_snapshot.go (hunk 2 above)
Comment (hairyhum) [reply pkg/function/create_volume_snapshot.go:137]: Or maybe errkit, but I was trying to keep the changes minimal.

File: pkg/function/delete_rds_snapshot.go (hunk 1 above)
Comment (hairyhum) [reply pkg/function/delete_rds_snapshot.go:58]: Upgrading linter version is a github action, which I've already run, so it's just a matter of getting it to the `master` codebase

File: pkg/function/create_volume_snapshot.go (hunk 2 above)
Comment (hairyhum) [reply pkg/function/create_volume_snapshot.go]: Changed to errkit because we already have it in this file.

**Pull request merged**